### ISSUE TITLE
[BUG] Proform Pro 2000 Not Connecting

### ISSUE THREAD

**Issue by bagspacked** [cagnulein/QZCompanionNordictrackTreadmill#118]
Hi,
I have a Proform Pro 2000.  I completed the setup via ADB and it worked.  I'm a little confused about which options to select on the treadmill and in the app, there is a Proform (not Pro) 2000 option.  I have tried a bunch of different settings and can't seem to get the tiles to update no matter what I try.  The tiles don't update.  I am putting the treadmill IP in the Nordic Track (not Pro Form) box.  Let me know if there's a fix I might try.
Thanks

**Comment by cagnulein** [comment]
Hi send me a debug log and I will check your setup
https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 19 ott 2024 alle 21:26 bagspacked ***@***.***>
ha scritto:

**Comment by bagspacked** [comment]
Thanks for your quick response. Here is the debug log
[debug-Sat_Oct_19_15_06_48_2024.log](https://github.com/user-attachments/files/17447211/debug-Sat_Oct_19_15_06_48_2024.log)

**Comment by cagnulein** [comment]
I can see the companion sending frames but they are all empty. Try to set
the c1750 on the companion and restart the treadmill

Let me know

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno sab 19 ott 2024 alle 22:12 bagspacked ***@***.***>
ha scritto:

**Comment by bagspacked** [comment]
Hi,
I have tried c1750 and restarted but I still have the same result-  no data
on the tiles. Please advise and thanks!

**Comment by cagnulein** [comment]
Please share always a debug log, so I can check if there is a difference,
we have to find the right profile

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 20 ott 2024 alle 14:06 bagspacked ***@***.***>
ha scritto:

**Comment by bagspacked** [comment]
Ah of course here you go. I tried both the 1750 and 1750 2020 profile as I believe my treadmill is from 2020
[debug-Sun_Oct_20_06_55_03_2024.log](https://github.com/user-attachments/files/17450295/debug-Sun_Oct_20_06_55_03_2024.log)

**Comment by cagnulein** [comment]
ok this is worst.
let's try this version that use OCR
https://github.com/cagnulein/QZCompanionNordictrackTreadmill/actions/runs/11010660989/artifacts/1970559335
you will see a popup asking for recording the screen, say yes
let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 20 ott 2024 alle ore 14:48 bagspacked <
***@***.***> ha scritto:

**Comment by bagspacked** [comment]
Ok, sorry I am not good at programming or software.  How do I get this on
the treadmill doing ADB?  Do I need to send a terminal command?

Thanks

**Comment by cagnulein** [comment]
how did you install it the actual qz companion version there?

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 20 ott 2024 alle ore 15:38 bagspacked <
***@***.***> ha scritto:

**Comment by bagspacked** [comment]
I used the ADB method via macOS, by running the install package

**Comment by cagnulein** [comment]
ok perfect so replace the apk with the one that I just provided there and
rerun the install script

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 20 ott 2024 alle ore 15:40 bagspacked <
***@***.***> ha scritto:

**Comment by bagspacked** [comment]
Excellent, I can now see the relevant data on the tiles.  With this connection method, is it possible to control the treadmill remotely as well (from the QZ app or a third party app)?
Thank you for your help

**Comment by bagspacked** [comment]
Also, the output seems to be reversed between km and miles.  In other words, my treadmill is reading out miles per hour but the app seems to think they are in km.  I have tried to adjust the toggle but either way it doesn't report correctly.  
Thanks once again for helping to troubleshoot.

**Comment by cagnulein** [comment]
Yes, you can do that.
I just need a screenshot of the manual workout window of ifit using the adb
screenshoot script!

Let me know!

Il giorno dom 20 ott 2024 alle 22:25 bagspacked ***@***.***>
ha scritto:

**Comment by bagspacked** [comment]
Hi, I have tried to run the adb-screenshot-mac.sh by typing bash and the script address.  When I do this it asks for the treadmill IP, then it runs pings successfully, but then it it says no such file or directory.  Am I following the correct procedure?

**Comment by cagnulein** [comment]
Please share me the full log of this operation

I'm going to bed now we will continue tomorrow!

Anyway we did a huge step forward!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 20 ott 2024 alle 23:00 bagspacked ***@***.***>
ha scritto:

**Comment by bagspacked** [comment]
Ok, the prior error was my mistake in how I was executing the command in Terminal.  Now I fixed that, but it doesn't seem able to connect to the treadmill.


```
priggy@MacBook-Air ~ % cd /Users/priggy/Downloads/QZCompanionNordictrackTreadmill-master/InstallPackage/                  
priggy@MacBook-Air InstallPackage % bash adb-screenshot-mac.sh
Enter treadmill IP address: 192.168.1.53
PING 192.168.1.53 (192.168.1.53): 56 data bytes
64 bytes from 192.168.1.53: icmp_seq=0 ttl=64 time=27.585 ms
64 bytes from 192.168.1.53: icmp_seq=1 ttl=64 time=15.324 ms
64 bytes from 192.168.1.53: icmp_seq=2 ttl=64 time=17.772 ms
64 bytes from 192.168.1.53: icmp_seq=3 ttl=64 time=11.536 ms

--- 192.168.1.53 ping statistics ---
4 packets transmitted, 4 packets received, 0.0% packet loss
round-trip min/avg/max/stddev = 11.536/18.054/27.585/5.934 ms
disconnected everything
* daemon not running; starting now at tcp:5037
* daemon started successfully
failed to connect to '192.168.1.53:5555': Connection refused
List of devices attached

adb: no devices/emulators found
adb: error: failed to get feature set: no devices/emulators found
adb: no devices/emulators found
priggy@MacBook-Air InstallPackage % 
```

**Comment by cagnulein** [comment]
Hi @bagspacked i created a new version of the script, you can find it here

https://raw.githubusercontent.com/cagnulein/QZCompanionNordictrackTreadmill/refs/heads/master/InstallPackage/adb-screenshot-mac.sh

let's see if it's better, i used the same behaviour of the installer, so it should work
Let me know

**Comment by bagspacked** [comment]
I'm still getting the same issue.  Here is the Terminal report

Last login: Sun Oct 20 16:04:10 on ttys001

***@***.*** ~ % cd
/Users/priggy/Downloads/QZCompanionNordictrackTreadmill-master/InstallPackage

***@***.*** InstallPackage % bash adb screenshot mac.sh

adb: adb: cannot execute binary file

***@***.*** InstallPackage % bash adb-screenshot-mac.sh

Enter treadmill IP address: 192.168.1.53

PING 192.168.1.53 (192.168.1.53): 56 data bytes

64 bytes from 192.168.1.53: icmp_seq=0 ttl=64 time=134.961 ms

64 bytes from 192.168.1.53: icmp_seq=1 ttl=64 time=48.083 ms

64 bytes from 192.168.1.53: icmp_seq=2 ttl=64 time=67.722 ms

64 bytes from 192.168.1.53: icmp_seq=3 ttl=64 time=192.265 ms


--- 192.168.1.53 ping statistics ---

4 packets transmitted, 4 packets received, 0.0% packet loss

round-trip min/avg/max/stddev = 48.083/110.758/192.265/57.029 ms

Cannot establish ADB connection.

***@***.*** InstallPackage %

**Comment by cagnulein** [comment]
no it's a different error. it seems that you didn't put the file into the installation folder because it can't find adb
let me know

**Comment by bagspacked** [comment]
Hi, Please see the following logs.  I did have it in the wrong directory
initially, but I've fixed that.  Now it's having an issue connecting. The
log is showing connection refused:

 First here is what I see in terminal:

Last login: Thu Oct 24 18:40:28 on console

***@***.*** ~ % cd
/Users/priggy/Downloads/QZCompanionNordictrackTreadmill-master/InstallPackage

***@***.*** InstallPackage % bash adb-screenshot-mac.sh

Enter treadmill IP address: 192.168.1.53

PING 192.168.1.53 (192.168.1.53): 56 data bytes

64 bytes from 192.168.1.53: icmp_seq=0 ttl=64 time=69.121 ms

64 bytes from 192.168.1.53: icmp_seq=1 ttl=64 time=87.677 ms

64 bytes from 192.168.1.53: icmp_seq=2 ttl=64 time=214.238 ms

64 bytes from 192.168.1.53: icmp_seq=3 ttl=64 time=132.449 ms


--- 192.168.1.53 ping statistics ---

4 packets transmitted, 4 packets received, 0.0% packet loss

round-trip min/avg/max/stddev = 69.121/125.871/214.238/55.972 ms

Cannot establish ADB connection.

***@***.*** InstallPackage %


In the qz companion log I see:


disconnected everything
* daemon not running; starting now at tcp:5037
* daemon started successfully
failed to connect to '192.168.1.53:5555': Connection refused

**Comment by cagnulein** [comment]
yeah seems so. what do you change from the time that you are able to install the companion to now? i mean adb was working at the beginning but it seems that it stops then. We have to figure out this, but unfortunately only you can understand if there is something that happens that close the adb port. @victorypoint do you have any idea?

**Comment by bagspacked** [comment]
I could try resetting the treadmill, running the install and then
immediately running the screenshot tool?

The only thing that I have changed is picking the treadmill type in the
companion screen and rebooting the treadmill.

Also, is there something I can change on the router to keep the port open?

**Comment by cagnulein** [comment]
or you can simply get the screenshot first and then installing the apk. it
should work

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno ven 25 ott 2024 alle ore 15:33 bagspacked <
***@***.***> ha scritto:

**Comment by mong00se007** [comment]
Hi Roberto, 
Just got my Proform Pro 2000, and was looking at a way to get run data into home assistant, and this seems like the best solution. I receive the the data in the app o.k but speed and incline cant be adjusted. however when I select incline it does pause the workout, so the feed back is there. Anyway I have attached two screenshots of the display. (as that is what looked like this thread was trying to do. )

In the app under specific model I left it at Nordictrack S25 and filled in the IP address under Nordictrack 2950 IP:

Thanks for making an awesome app :)
![06-11-2024@14-50-34 14](https://github.com/user-attachments/assets/9787e297-ec2d-40f0-aba3-3d5df7f07a8c)
![06-11-2024@14-52-29 12](https://github.com/user-attachments/assets/35420128-4e78-4283-a23a-5e7ad44de2f1)

**Comment by cagnulein** [comment]
Thanks I will add these ASAP!

**Comment by mong00se007** [comment]
thank you!

**Comment by cagnulein** [comment]
hi @mong00se007 sorry for the delay. I just checked your screenshot and the proform 200 device in the companion should be already right for your device. did you try it? if it wasn't set to this before, set it and reboot the treadmill
let me know!

**Comment by mong00se007** [comment]
Hi Roberto,
So I tried setting it to Proform 2000 (not pro) 
I then tried putting the ip under the Proform, but then it would not connect, so I set it back under the Nordictrack 2950 IP and then it does connect. 
It updates when I adjust on the treadmill in the app
but not when I change the speed in the app back on the treadmill.
ADB Remote is on.

Im using Android and the app version is 2.16.66

Thank you :)

**Comment by cagnulein** [comment]
But do you see a reaction on the touchscreen of the treadmill when you try to change speed from qz?

**Comment by mong00se007** [comment]
It responds when I do a height adjustment on the app. similar to just touching the middle of the screen, when I do a speed adjustment in the app it does nothing.

Thank you.

**Comment by cagnulein** [comment]
Hah i guess I understood the issue. Disable the adb remote and restart qz and let me know!

**Comment by mong00se007** [comment]
Hi,
Still the same, restarted the app and the treadmill afterwards, when I click on the speed it does nothing and incline is a touch on the screen but does not control it. both up and down touch on the same area of the screen. (managed to enable show touches on the screen so see the dot below the incline, in the photo)
![20241117_115627](https://github.com/user-attachments/assets/8d78fc81-db4b-4436-a661-2cd78384ee36)

**Comment by cagnulein** [comment]
Can you collect the adb logcat from the tablet while you are doing this? I guess there is a missing piece because it should already work

**Comment by mong00se007** [comment]
I'll do it in the morning, just to check I just download the logs through the app? 
as in https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

I also saw that the Nordictrack T6.5 has a 10" screen so ill try that as well. 
Thank you for your patience :)

**Comment by cagnulein** [comment]
You can start with the log from the qz app but I like more the log from the companion. There is a debug log script in the installation folder to get it

Let me know!

**Comment by mong00se007** [comment]
Changing to T6.5 didnt make a difference. I also tried changing the screen size in settings, as it wasnt on default but that did not help as far as I know, please find attached the logs. at 10:07 I clicked on the adjustment a few times.

P.S Just saw the interview you did https://www.youtube.com/watch?v=30hzZM3ETqQ very insperational, a feature I definetly will be using is the adjusting of speed acording to heart rate, as I like to run fast, but currently am not fit enough to do so. :)
[2024-11-18.logs.txt](https://github.com/user-attachments/files/17792759/2024-11-18.logs.txt)
[2024-11-18_logs.txt](https://github.com/user-attachments/files/17792760/2024-11-18_logs.txt)

Thank you, 
Lawrence

**Comment by cagnulein** [comment]
hi @mong00se007 i checked the log but there is no sign of QZ companion at all there! so strange: the only possibility is, where you still on the manual workout when you collect the logs? because ifit could remove the logs when you press stop on the manual workout.
Let me know

**Comment by Wilson-Micah** [comment]
Hi @cagnulein I picked this up yesterday to see if I could get working with my ProForm Pro 2000 treadmill (2018 version with 7' screen) and ran into similar issues as reported here. Was unable to get any data sending with original version, but after installing the OCR version I am able to see data being sent to the QZ app. 

Two issues I'm seeing: 

1. After pulling a screenshot and looking through the source code I found that the exp7i was going to be closest in lining up the pixels for slider touches. The speed will increase, but incline seems to always decrease I assume because my treadmill has 15% incline and the exp7i only has 10 so the positions are slightly off. I'll attach a screenshot.

2. The OCR companion app does seem to always assume the treadmill is in metric. I set it to metric to get the data correct and take a screenshot but I'd prefer imperial. I may try and modify the code and try and fix these issues myself as I am a software engineer although not an Android one. I would appreciate it if you can guide me in the right direction to fixing this.

![2024-12-02@08-35-58](https://github.com/user-attachments/assets/bdd62442-8aa7-4775-9f75-14678b97acf3)

**Comment by cagnulein** [comment]
> 2\. The OCR companion app does seem to always assume the treadmill is in metric. I set it to metric to get the data correct and take a screenshot but I'd prefer imperial. I may try and modify the code and try and fix these issues myself as I am a software engineer although not an Android one. I would appreciate it if you can guide me in the right direction to fixing this.

sure! it's very easy. we can do even automatically! if you see the MPH instead of KPH we can send the value divided by the miles/km constant to the QZ app (qz is always in km/h in the lower layer)



> 1. After pulling a screenshot and looking through the source code I found that the exp7i was going to be closest in lining up the pixels for slider touches. The speed will increase, but incline seems to always decrease I assume because my treadmill has 15% incline and the exp7i only has 10 so the positions are slightly off. I'll attach a screenshot.

great! can you do a PR changing the equation in the UDPListenerService.java 

you can fork this https://github.com/cagnulein/QZCompanionNordictrackTreadmill/pull/110

Let me know!

**Comment by Wilson-Micah** [comment]
@cagnulein I ended up making the changes and got it mostly working however the OCR version wasn't reliable enough for me to continue moving forward with it. For example when setting the speed to 7.0 the OCR would usually come back with 1.0 and would frequently match 8's to 0's. Common problem with OCR but had enough issues that I gave up on it even though I got the MPH sending correctly and the slider position in a good spot for adjusting speed and grade from the QZ app.

After this I went digging to see if I could figure out where treadmill data is being logged to in the current iFit app to try and go back to the regular version of the companion app. After looking through every log file it seems like speed and grade changes are no longer being logged on the latest version of the iFit app, at least nowhere I could find. 

I've ended up just installing an old version of the iFit app and the latest companion app works fine with that.
